sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala

```diff
@@ -64,7 +64,9 @@ case class SortOrder(
     direction: SortDirection,
     nullOrdering: NullOrdering,
     sameOrderExpressions: Set[Expression])
```

Comment:
We need to use `Set` for this variable?

Reply:
The set property of this variable is used at couple of places in SortMergeJoin ([here](https://github.com/apache/spark/blob/3fdfce3120f307147244e5eaf46d61419a723d50/sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala#L122), [here](https://github.com/apache/spark/blob/v3.0.0/sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala#L149)).

We can use ExpressionSet maybe at those 2 places and make it a Seq in constructor here. any thoughts?

Reply:
> We can use ExpressionSet maybe at those 2 places and make it a Seq in constructor here. any thoughts?

Yea, I think it is okay to just deduplicate it before storing it in the class as you said, and then make it `Seq`.